internal/pkg/gateway/api.go

```diff
@@ -273,6 +270,44 @@ func (gs *Server) Endorse(ctx context.Context, request *gp.EndorseRequest) (*gp.
 	return endorseResponse, nil
 }
 
+// responseStatus unpacks the proposal response and error values that are returned from ProcessProposal and
+// determines how the gateway should react (retry?, close connection?).
+// Uses the grpc canonical status error codes and their recommended actions.
+// Returns:
+// - response successful (bool)
+// - error message extracted from the err or generated from 500 proposal response
+// - should the gateway retry (only the Evaluate() uses this)
+// - should the gateway close the connection and remove the peer from its registry
```

Comment:
What exactly does remove from registry mean? Just until the next time we check with discovery? Or longer?

Reply:
The former.  It closes the connection and removes from the registry, but if the next call to discovery returns that peer in its plan, then it will attempt to reconnect and add it back.  This could happen if the peer had gone down, but the gossip has not notified the discovery service yet.

Reply:
Ok, I would mention this in the removeEndorser() function comment.